.github/workflows/ci.yml

```diff
@@ -0,0 +1,30 @@
+name: ci
+
+on:
+  push:
+    branches:
+      - "master"
+  pull_request: {}
+
+jobs:
+  build:
+    name: "CI"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Build Docker image
+        run: |
+          docker build . -t gnssrefl
+      - name: Create Docker container
+        run: |
+          docker run --name gnssrefl -d -v "$PWD:/src" gnssrefl tail -f /dev/null
+      - name: Run smoke tests
+        run: |
+          docker exec -i gnssrefl /src/software_tests
+      - name: Install pytest
+        run: |
+          docker exec -i gnssrefl pip3 install 'pytest~=7.2' 'pytest-mock~=3.10'
```

Comment:
There might be a way to specify development dependencies in `setup.py`, but this was easy so I did it like this. Provide version constraints on the dependencies so we don't get breaking changes unexpectedly.